src/AppInstallerCommonCore/Fonts.cpp

```diff
@@ -0,0 +1,188 @@
+
+            wil::com_ptr<IDWriteFontFile> fontFiles[8];
+            THROW_HR_IF(E_OUTOFMEMORY, fileCount > ARRAYSIZE(fontFiles));
+            THROW_IF_FAILED(fontFace->GetFiles(&fileCount, fontFiles[0].addressof()));
+
+            std::vector<std::filesystem::path> filePaths;
+            for (UINT32 i = 0; i < fileCount; ++i) {
+                wil::com_ptr<IDWriteFontFileLoader> loader;
+                THROW_IF_FAILED(fontFiles[i]->GetLoader(loader.addressof()));
+
+                const void* fontFileReferenceKey;
+                UINT32 fontFileReferenceKeySize;
+                THROW_IF_FAILED(fontFiles[i]->GetReferenceKey(&fontFileReferenceKey, &fontFileReferenceKeySize));
+
+                if (const auto localLoader = loader.try_query<IDWriteLocalFontFileLoader>()) {
+                    UINT32 pathLength;
+                    THROW_IF_FAILED(localLoader->GetFilePathLengthFromKey(fontFileReferenceKey, fontFileReferenceKeySize, &pathLength));
+                    pathLength += 1; // Account for the trailing null terminator during allocation.
+
+                    wchar_t path[MAX_PATH];
+                    THROW_HR_IF(E_OUTOFMEMORY, pathLength > ARRAYSIZE(path));
+                    THROW_IF_FAILED(localLoader->GetFilePathFromKey(fontFileReferenceKey, fontFileReferenceKeySize, &path[0], pathLength));
+                    std::filesystem::path fontFilePath = { AppInstaller::Utility::Normalize(std::wstring(path)) };
+                    filePaths.push_back(std::move(fontFilePath));
+                }
+            }
+
+            return std::move(filePaths);
+        }
+
+        std::wstring GetFontFaceName(const wil::com_ptr<IDWriteFont>& font)
+        {
+            wil::com_ptr<IDWriteLocalizedStrings> faceNames;
+            THROW_IF_FAILED(font->GetFaceNames(faceNames.addressof()));
+
+            wchar_t localeNameBuffer[LOCALE_NAME_MAX_LENGTH];
+            const auto localeName = GetUserDefaultLocaleName(localeNameBuffer, LOCALE_NAME_MAX_LENGTH) ? localeNameBuffer : L"en-US";
+
+            UINT32 faceNameIndex;
+            BOOL faceNameExists;
+            if (FAILED(faceNames->FindLocaleName(localeName, &faceNameIndex, &faceNameExists)) || !faceNameExists)
+            {
+                faceNameIndex = 0;
+            }
+
+            UINT32 faceNameLength = 0;
+            THROW_IF_FAILED(faceNames->GetStringLength(faceNameIndex, &faceNameLength));
+            faceNameLength += 1; // Account for the trailing null terminator during allocation.
+
+            wchar_t faceName[32]; // Max size for font name including null terminator is 32.
+            THROW_HR_IF(E_OUTOFMEMORY, faceNameLength > ARRAYSIZE(faceName));
+            THROW_IF_FAILED(faceNames->GetString(faceNameIndex, &faceName[0], faceNameLength));
+            return std::wstring(faceName);
+        }
+
+        std::wstring GetFontFamilyName(const wil::com_ptr<IDWriteFontFamily>& fontFamily)
```

Comment:
This is exactly the same function as `GetFontFaceName` save for the input type and function that is called to get the `IDWriteLocalizedStrings`.  Move all of the "get the correct localized string from an `IDWriteLocalizedStrings`" to a shared function.